text/0148-sampling-probability.md

```diff
@@ -0,0 +1,291 @@
+itself has occurred _adjusted count_ times.  Every sample event is
+representative for adjusted count many copies of itself.
+
+There is one essential requirement for this to work.  The selection
+procedure must be _statistically unbiased_, a term meaning that the
+process is required to give equal consideration to all possible
+outcomes.
+
+### Encoding inclusion probability
+
+Some possibilities for encoding the inclusion probability, depending
+on the circumstances and the protocol, briefly discussed:
+
+1. Encode the adjusted count directly as a floating point or integer number in the range [0, +Inf).  This is a conceptually easy way to understand sampling because larger numbers mean greater representivity.
+2. Encode the inclusion probability directly as a floating point number in the range [0, 1).  This is typical of the Statsd format, where each line includes an optional probability.  In this context, the probability is commonly referred to as a "sampling rate".  In this case, smaller numbers mean greater representivity.
+3. Encode the negative of the base-2 logarithm of inclusion probability.  This restricts inclusion probabilities to powers of two and allows the use of small non-negative integers to encode power-of-two adjusted counts.
+4. Fold the adjusted count into the data.  This is appropriate when the data itself carries counts, such as for OTLP Metrics Sum and Histogram points encoded using delta aggregation temporality.  This may lead to rounding errors, when adjusted counts are not integer valued.
+
+This is not an exhaustive list of approaches.  All of these techniques
+are considered appropriate.
+
+A telemetry system should be able to accurately estimate the number of
+events that took place whether the events were sampled or not, which
+requires being able to recognize a zero value for the adjusted count
+as being distinct from a raw event where no sampling took place.
+
+#### Recognizing zero adjusted count
+
+An adjusted count of zero indicates an event that was recorded, where
+according to the sampling design its inclusion probability is zero.
+These events are may be included in a stream of sampled events as
+auxiliary information, and consu
+
+Recording events with zero adjusted count are a useful way to record
+auxiliary information while sampling, events that are considered
+interesting but which are accounted for by the adjusted count of other
+events in the same stream.
+
+Consider a span sampling design that applies a sampling decision only
+to the roots of a trace.  Non-root spans must be recorded when their
+parent span is selected for a sample.  For a class of spans that
+sometimes are and sometimes are not the root of a trace, we have three
+outcomes:
+
+1. Span is part of another trace: adjusted count is zero
+2. Span is a root, was selected for the sample: adjusted count is non-zero
+3. Span is a root, was not selected for the sample: not recorded.
+
```

Comment:
What I would expect, when aggregating the _adjusted counts_ of all spans of one particular service, is an unbiased estimate of the true number of requests to that service. Unfortunately this is not the case, if there are non-root spans for that particular service, because the sampling decision is made at root spans, and the corresponding sample rates and hence _adjusted counts_ are not known for non-root spans. Consequently, a zero _adjusted count_ would just indicate that the _adjusted count_ is not known and needs to be retrieved from the root span. However, it does not necessarily mean that the span does not contribute to the total count at all.

Therefore, I would prefer to propagate the sample rate that was used for some root span to its descendants, such that they know the sample rate and are able to set the _adjusted count_ properly.

Reply:
I was trying to illustrate an example, not to suggest a real tracing Sampler, however I can see that no one is interested in this case, so it's not a good example.

I removed these paragraphs. Instead, I added more depth to the description of the Dapper system, and how it propagated the parent's inclusion probability, particularly when it made a negative sampling decision, so the child could calculate the conditional probability needed to ensure an overall rate of its own traces. I sense there is more interested in this approach, now that tracing has reached GA.